python/ray/autoscaler/command_runner.py

```diff
@@ -198,6 +202,10 @@ def run_rsync_down(self, source, target):
         if target.startswith("~"):
             target = "/root" + target[1:]
 
+        # Add trailing slashes for rsync
+        source += "/" if not source.endswith("/") else ""
+        target += "/" if not target.endswith("/") else ""
```

Comment:
I think this might not work though; the reason is because you can rsync individual files too

Reply:
Right. I thought about adding a directory check for the local directory, but this doesn't work as rsync can also sync to not-yet-existing directories. So I'll move it back to the KubernetesSyncClient, as that one is only called with directories